_posts/setting-up-natty.md

```diff
@@ -0,0 +1,297 @@
+
+After making those changes, you'll need to run this command:
+
+    mvn clean package
+
+That should look like `ubuntu@ubuntu:~/Natty$ mvn clean package`.
+
+Your terminal will tell you what it's doing again, which you probably won't be very interested in reading.
+
+You might get an error at this point:
+
+    COMPILATION ERROR : 
+    [INFO] -------------------------------------------------------------
+    [ERROR] No compiler is provided in this environment. Perhaps you are running on a JRE rather than a JDK?
+
+To solve this, you have to run this command and install the JDK:
+
+    sudo apt-get install openjdk-8-jdk
+
+Now you have to set up the folders and files in the "target" folder. Go to the file system, and find Home > Natty > target. Right click, and create a new folder called "data". So you should have Home > Natty > target > data.  Now you can open up the text editor (press the <kbd>super</kbd> key, type `gedit`, and open it up), and create the following blank files and save them in "data": `BlackListedUsers.txt`, `BlackListedWords.txt`, `CheckUsers.txt`, `FeatureRequests.txt`, `OptedInUsersList.txt`, `Salutations.txt`, and `WhiteListedWords.txt`. 
+
+Now, some of these we're going to fill, but most of these we're going to leave blank.
+
+We're going to fill in `BlackListedWords.txt` first. What goes in here are the keywords that will make Natty detect the post, and there's a list of keywords maintained by SOBotics, that are available from the API at https://logs.sobotics.org/napi/api/list/blacklistedWords.
+
+The file needs to be set up so that the list of keywords looks like this:
+
+    could someone explain
+    can anyone give me an answer
+    Please help me in resolving
+    Stackoverflow won't let me comment
+    Has this been an unresolved issue
+    etc...
+
+If you don't care about having the most recent data, you can copy-paste [this pastebin](https://pastebin.com/raw/WDCJAUBu), from March 19, 2018.
+
+Next we fill in 'WhiteListedWords.txt`. Like the blacklist, you can fetch it from the API: https://logs.sobotics.org/napi/api/list/whitelistedWords
+
+This is arranged in the same way as the blacklist:
+
+    Hope this helps
+    May it help someone
+    problem disappears
+    etc...
+
+For convenience, [here](https://pastebin.com/raw/5KfGZfYq) a pastebin of the whitelist as well, from March 19, 2018.
+
+Next we do `Salutations.txt`. There is no API option for salutations, but @BhargavRao provided his list in [chat](https://chat.stackoverflow.com/transcript/message/41703588#41703588). This should be in the same format as the blacklist and whitelist.
```

Comment:
"but you can use [this list of salutations](https://chat.stackoverflow.com/message/41703588?plain=true)". Trying to make it like a blog instead of an answer.